controllers/autoneg_test.go

```diff
@@ -297,6 +309,125 @@ func TestValidateConfig(t *testing.T) {
+	tests := []struct {
+		name                   string
+		config                 AutonegConfig
+		err                    bool
+		expectedCapacityScaler float64
+	}{
+		{
+			name:                   "default config",
+			config:                 AutonegConfig{},
+			err:                    false,
+			expectedCapacityScaler: 1,
+		},
+		{
+			name: "negative initial_capacity",
+			config: AutonegConfig{
+				BackendServices: map[string]map[string]AutonegNEGConfig{
+					"80": {
+						"http-be": {
+							Name:            "http-be",
+							Connections:     100,
+							InitialCapacity: pointer.Int32Ptr(int32(-10)),
+						},
+					},
+				},
+			},
+			err:                    true,
+			expectedCapacityScaler: 1,
+		},
+		{
+			name: "large initial capacity",
+			config: AutonegConfig{
+				BackendServices: map[string]map[string]AutonegNEGConfig{
+					"80": {
+						"http-be": {
+							Name:            "http-be",
+							Connections:     100,
+							InitialCapacity: pointer.Int32Ptr(int32(5000)),
+						},
+					},
+				},
+			},
+			err:                    true,
+			expectedCapacityScaler: 1,
+		},
+		{
+			name: "zero initial capacity",
+			config: AutonegConfig{
+				BackendServices: map[string]map[string]AutonegNEGConfig{
+					"80": {
+						"http-be": {
+							Name:            "http-be",
+							Connections:     100,
+							InitialCapacity: pointer.Int32Ptr(int32(0)),
+						},
+					},
+				},
+			},
+			err:                    false,
+			expectedCapacityScaler: 0,
+		},
+		{
+			name: "half initial capacity",
+			config: AutonegConfig{
+				BackendServices: map[string]map[string]AutonegNEGConfig{
+					"80": {
+						"http-be": {
+							Name:            "http-be",
+							Connections:     100,
+							InitialCapacity: pointer.Int32Ptr(int32(50)),
+						},
+					},
+				},
+			},
+			err:                    false,
+			expectedCapacityScaler: 0.5,
+		},
+		{
+			name: "max initial capacity",
+			config: AutonegConfig{
+				BackendServices: map[string]map[string]AutonegNEGConfig{
+					"80": {
+						"http-be": {
+							Name:            "http-be",
+							Rate:            100,
+							InitialCapacity: pointer.Int32Ptr(int32(100)),
+						},
+					},
+				},
+			},
+			err:                    false,
+			expectedCapacityScaler: 1,
+		},
+	}
+
+	for _, ct := range tests {
+		err := validateNewConfig(ct.config)
+		if err == nil && ct.err {
+			t.Errorf("Set %q: expected error, got none", ct.name)
+		}
+		if err != nil && !ct.err {
+			t.Errorf("Set %q: expected no error, got one: %v", ct.name, err)
+		}
+
+		// The compute.Backend object should have a float64 value in
+		// the range [0.0, 1.0]
+		status := AutonegStatus{AutonegConfig: ct.config}
+		beConfig := status.Backend("http-be", "80", "group")
+		if beConfig.CapacityScaler < 0 || beConfig.CapacityScaler > 1 {
+			t.Errorf("Set %q: expected capacityScaler in [0.0, 1.0], got %f", ct.name, beConfig.CapacityScaler)
+		}
+
+		// Actual value should be within 1e-9 of expected
```

Comment:
Added this check that our values are +/- 1e-9 of our expected value in different situatiosn